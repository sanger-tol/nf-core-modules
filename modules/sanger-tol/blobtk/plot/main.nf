process BLOBTK_PLOT {
    // Linked to issue https://github.com/sanger-tol/genomenote/issues/184
    // Depending on the blob dataset in use, the grid option may not
    // work at all. This is down to the version of blobtoolkit used to
    // generate the blob.
    // Adding a check would overly complicate the module so for now
    // we can ignore errors, with the *assumption* it would only kill
    // runs in which the blobdir doesn't have the right data
    errorStrategy = 'ignore'

    tag "$blobtk_args.name"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "community.wave.seqera.io/library/blobtk:0.7.1--e3f63bb2cdc8fb96"

    input:
    tuple val(meta), path(fasta)
    path(local_path)    // Genuine path location must be a path.
    val(online_path)    // HTTPS location needs to remain a value
    each blobtk_args    // "each" so that we can easily run the module with different parameters

    output:
    tuple val(meta), path("*.png"), emit: png
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args         = task.ext.args ?: ''
    def prefix       = task.ext.prefix ?: "${meta.id}_${blobtk_args.name}"
    def resource     = online_path ?: local_path
    """
    blobtk plot \\
        -d $resource \\
        $blobtk_args.args \\
        $args \\
        -o ${prefix}.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        blobtk: \$(blobtk --version | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix       = task.ext.prefix ?: "${meta.id}_${blobtk_args.name}"
    """
    touch ${prefix}.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        blobtk: \$(blobtk --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
