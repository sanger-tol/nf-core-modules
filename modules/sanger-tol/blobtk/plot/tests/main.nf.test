nextflow_process {

    name "Test Process BLOBTK_PLOT"
    script "../main.nf"
    process "BLOBTK_PLOT"

    tag "modules"
    tag "modules_sangertol"
    tag "blobtk"
    tag "blobtk/plot"
    tag "nf-core/gunzip/gunzip"

    setup {
        nfcoreInitialise("${launchDir}/library/")
        nfcoreInstall("${launchDir}/library/", ["gunzip"])
        nfcoreLink("${launchDir}/library/", "${baseDir}/modules/")

        run("GUNZIP") {
            script "../../../../nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id: "test" ],
                    file(params.modules_testdata_base_path + 'Ceramica_pisi/assembly/release/ilCerPisi1.1/insdc/GCA_963859965.1.fasta.gz', checkIfExists: true)
                ])
                """
            }
        }
    }

    config './nextflow.config'

    test("Ceramica pisi - fasta") {

        when {

            process {
                """
                input[0] = GUNZIP.out.gunzip

                input[1] = []

                input[2] = "https://blobtoolkit.genomehubs.org/api/v1/dataset/id/CALUEP01"

                input[3] = [
                    [
                        name: "BLOB_VIEW",
                        args: "-v blob"
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Ceramica pisi - fasta - stub") {

        options "-stub"

        when {

            process {
                """
                input[0] = GUNZIP.out.gunzip

                input[1] = []

                input[2] = "https://blobtoolkit.genomehubs.org/api/v1/dataset/id/CALUEP01"

                input[3] = [
                    [
                        name: "BLOB_VIEW",
                        args: "-v blob"
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
