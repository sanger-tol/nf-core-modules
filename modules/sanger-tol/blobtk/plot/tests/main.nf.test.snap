{
    "Ceramica pisi - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_BLOB_VIEW.png:md5,39fa1e631e10d8a8c2034ac79cde3fc0"
                    ]
                ],
                "1": [
                    "versions.yml:md5,809b40a0db8355fe992de6daf2abf4d3"
                ],
                "png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_BLOB_VIEW.png:md5,39fa1e631e10d8a8c2034ac79cde3fc0"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,809b40a0db8355fe992de6daf2abf4d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.2"
        },
        "timestamp": "2025-09-03T13:02:05.425828"
    },
    "Ceramica pisi - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_BLOB_VIEW.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,809b40a0db8355fe992de6daf2abf4d3"
                ],
                "png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_BLOB_VIEW.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,809b40a0db8355fe992de6daf2abf4d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.2"
        },
        "timestamp": "2025-09-03T13:02:30.841068"
    }
}